

\subsection{Detailed Guide}

\begin{enumerate}
 \item Add the input protein fasta file to \mintinline{bash}{input/raw/}
 \item Make a copy of the \mintinline{bash}{config.json} file with a different name (e.g.  \mintinline{bash}{test.json})
 \item Update \mintinline{bash}{test.json}
    \begin{enumerate}
        \item Update the \mintinline{bash}{input} section and make changes to match your input fasta and config file
        \item input section has the following parameters that can be set
        \begin{table}[H]
            \centering
            \caption{Parameters available to change in input section of the config file}
            \begin{tabular}{|l|l|}
              \hline
              \textbf{Parameter} & \textbf{Description} & \texbf{Example}
              \csvreader[head to column names]{tables/input_params.txt}{}%
              {\\\hline \Parameter & \Description & \Example }%
              \\ \hline
        \end{tabular}
      \end{table}
      
            % \begin{itemize}
            %     \item config\_file     : - The name of the current configuration  file
            %     \item work\_dir        : -
            %     \item gene\_start
            %     \item basename
            %     \item filt\_dir
            %     \item taxon
            %     \item raw\_fasta
            %     \item trans\_pattern
            %     \item version
            %     \item fasta
            %     \item log\_file
            %     \item inbred
            %     \item species
            % \end{itemize}
            
    \end{enumerate}
    
 \item Update the \mintinline{bash}{input} section

   \begin{itemize}

        \item
              Give the correct input FASTA file name
        \item
              If the fasta contains multiple transcripts per gene then put the
              fasta in the \texttt{input/raw} directory and set the
              \texttt{raw\_fasta} parameter
        \item
              If the fasta file contains only on transcript per gene put it in the
              \texttt{input/filt} directory, and set the \texttt{fasta} parameter
        \item
              Update the species, inbred and version parameters for your species
 \item
       {[}Optional{]} Update the \texttt{seq-sim} section

       \begin{itemize}

        \item
              (All the files should be already processed in this section)
       \end{itemize}
 \item
       {[}Optional{]} Update the \texttt{mix-meth} section

       \begin{itemize}

        \item
              (All the files and fields should be already set, except changes to
              database section for PANNZER )
       \end{itemize}
 \item
       {[}Optional{]} Update \texttt{blast} and \texttt{hmmer} sections

       \begin{itemize}

        \item
              This is to enable the correct number cpu threads for these software
       \end{itemize}
 \item
       All other sections should only be updated if things have been
       drastically changed.
\end{itemize}

 \item
       execute \texttt{python\ pipeline1.py\ config.json}
\end{enumerate}

\begin{itemize}

 \item
       The pipeline will generate a number of intermidiate output files
 \item
       Especially the mixed-method tools will require the input fasta to be
       split into smaller chunks. the chunks will be numbered serially.
       (e.g.~test.1.fa, test.2.fa)
 \item
       Argot 2.5 tool will NOT be executed within the pipeline
\end{itemize}

\begin{enumerate}
 \def\labelenumi{\arabic{enumi}.}
 \setcounter{enumi}{3}

 \item
       Submit the files in \texttt{mixed-meth/argot2.5/blast} and
       \texttt{mixed-meth/argot2.5/hmmer} using correct pairing
 \item
       Extract the Argot2.5 result files for each job, in the
       \texttt{mixed-meth/argot2.5/results} directory and rename with correct
       prefix
\end{enumerate}

\begin{itemize}

 \item
       Argot2.5 names all results as \texttt{argot\_results\_ts0.tsv} so the
       file should be renamed correctly (e.g.~test.1.tsv, test.2.tsv)
 \item
       Please do not leave any other file in the argot2.5 results directory,
       otherwise it will influence certain metrics.
\end{itemize}

\begin{enumerate}
 \def\labelenumi{\arabic{enumi}.}
 \setcounter{enumi}{5}

 \item
       execute \texttt{python\ pipeline2.py\ config.json}
\end{enumerate}



